packages/miniflare/src/workers/secrets-store/secret.worker.ts

```diff
@@ -0,0 +1,81 @@
+// Emulated Secret Store Binding
+
+import { WorkerEntrypoint } from "cloudflare:workers";
+import { ADMIN_API } from "./constants";
+
+// ENV configuration
+interface Env {
+	store: KVNamespace<string>;
+	secret_name: string;
+}
+
+export class SecretsStoreSecret extends WorkerEntrypoint<Env> {
+	async get(): Promise<string> {
+		const value = await this.env.store.get(this.env.secret_name, "text");
+
+		if (value === null) {
+			throw new Error(`Secret "${this.env.secret_name}" not found`);
+		}
+
+		return value;
+	}
+
+	[ADMIN_API]() {
+		return {
+			create: async (value: string) => {
+				const id = crypto.randomUUID().replaceAll("-", "");
+				await this.env.store.put(this.env.secret_name, value, {
+					metadata: { uuid: id },
+				});
+				return id;
+			},
+			update: async (value: string, id: string) => {
+				const { keys } = await this.env.store.list<{ uuid: string }>();
+				const secret = keys.find((k) => k.metadata?.uuid === id);
+				if (!secret) {
+					throw new Error(`Secret not found`);
+				}
+				await this.env.store.put(secret?.name, value, {
+					metadata: { uuid: id },
+				});
+				return id;
+			},
+			duplicate: async (id: string, newName: string) => {
+				const { keys } = await this.env.store.list<{ uuid: string }>();
+				const secret = keys.find((k) => k.metadata?.uuid === id);
+				if (!secret) {
+					throw new Error(`Secret not found`);
+				}
+				const existingValue = await this.env.store.get(secret.name);
+				if (!existingValue) {
+					throw new Error(`Secret not found`);
+				}
+				const newId = crypto.randomUUID();
+				await this.env.store.put(newName, existingValue, {
+					metadata: { uuid: newId },
+				});
+				return newId;
+			},
+			delete: async (id: string) => {
+				const { keys } = await this.env.store.list<{ uuid: string }>();
+				const secret = keys.find((k) => k.metadata?.uuid === id);
+				if (!secret) {
+					throw new Error(`Secret not found`);
+				}
+				await this.env.store.delete(secret?.name);
+			},
+			list: async () => {
+				const { keys } = await this.env.store.list<{ uuid: string }>();
+				return keys;
+			},
+			get: async (id: string) => {
```

Comment:
how is this `get` different from the one on https://github.com/cloudflare/workers-sdk/pull/8394/files#diff-98a047e6ac7c90b5177b6aaa402b11f25894a4864009ed9adb6779cb8269996eR13. Why do we need two?

Reply:
The above `get()` takes no arguments and is the API exposed to user workers (matching prod). This `get(id)` allows you to read _any_ secret (something not possible in prod), which is used by the `wrangler secrets-store secret get *` command